sap_hana/datadog_checks/sap_hana/sap_hana.py

```diff
@@ -596,3 +578,74 @@ def execute_query(self, cursor, query, source):
+
+            connection_properties.setdefault('address', self._server)
+            connection_properties.setdefault('port', self._port)
+            connection_properties.setdefault('user', self._username)
+            connection_properties.setdefault('password', self._password)
+
+            timeout_milliseconds = int(self._timeout * 1000)
+            connection_properties.setdefault('communicationTimeout', timeout_milliseconds)
+            connection_properties.setdefault('nodeConnectTimeout', timeout_milliseconds)
+
+            if self._use_tls:
+                connection_properties.setdefault('encrypt', True)
+                connection_properties.setdefault('sslHostNameInCertificate', self._server)
+                connection_properties.setdefault('sslSNIHostname', self._server)
+
+                tls_verify = self.instance.get('tls_verify', True)
+                if not tls_verify:
+                    connection_properties.setdefault('sslValidateCertificate', False)
+
+                tls_cert = self.instance.get('tls_cert')
+                if tls_cert:
+                    connection_properties.setdefault('sslKeyStore', tls_cert)
+
+                tls_ca_cert = self.instance.get('tls_ca_cert')
+                if tls_ca_cert:
+                    connection_properties.setdefault('sslTrustStore', tls_ca_cert)
+                elif not connection_properties.get('sslUseDefaultTrustStore', True):
+                    connection_properties.setdefault('sslTrustStore', certifi.where())
+
+            try:
+                connection = HanaConnection(**connection_properties)
+            except Exception as e:
+                error = str(e)
+                self.log.error('Unable to connect to SAP HANA: %s', error)
+                self.service_check(self.SERVICE_CHECK_CONNECT, self.CRITICAL, message=error, tags=self._tags)
+            else:
+                self.service_check(self.SERVICE_CHECK_CONNECT, self.OK, tags=self._tags)
+                return connection
+
+    else:
+
+        def get_connection(self):
+            try:
+                tls_context = self.get_tls_context() if self._use_tls else None
+                connection = HanaConnection(
+                    host=self._server,
+                    port=self._port,
+                    user=self._username,
+                    password=self._password,
+                    tls_context=tls_context,
+                    timeout=self._timeout,
+                )
+                connection.connect()
+            except Exception as e:
+                error = str(e)
+                self.log.error('Unable to connect to SAP HANA: %s', error)
+                self.service_check(self.SERVICE_CHECK_CONNECT, self.CRITICAL, message=error, tags=self._tags)
+            else:
+                self.service_check(self.SERVICE_CHECK_CONNECT, self.OK, tags=self._tags)
+                return connection
```

Comment:
Should this be in connection.py?

Reply:
No b/c I think it makes more sense for the split in config logic to be closer to the check since it needs attributes of the check